### PR TITLE
Sct corrupt diagramfiles

### LINKED ISSUE #2044: NPE when trying to open a corrupt .sct file
Whenever a .sct file is corrupt (i.e. containing <<< HEAD after a git-merge has been triggered) an NPE or SAXException occurs. Also if a valid .sct file has been changed to an invalid state and eclipse ask whether the updated content should be loaded due to the file changes the NPE or SAXException occurs.

image
image

We should handles this more graceful

stacktrace :
`org.eclipse.xtext.builder.resourceloader.IResourceLoader$LoadOperationException: org.xml.sax.SAXParseExceptionpublicId: platform:/resource/coffee_machine_pro/src/CoffeeMachine.sct; systemId: platform:/resource/coffee_machine_pro/src/CoffeeMachine.sct; lineNumber: 1; columnNumber: 2; Markup im Dokument vor dem Root-Element muss ordnungsgemäß formatiert sein.
	at org.eclipse.xtext.builder.resourceloader.SerialResourceLoader$1.next(SerialResourceLoader.java:50)
	at org.eclipse.xtext.builder.resourceloader.AbstractResourceLoader$CheckedLoadOperation.next(AbstractResourceLoader.java:77)
	at org.eclipse.xtext.builder.clustering.ClusteringBuilderState.writeNewResourceDescriptions(ClusteringBuilderState.java:382)
	at org.eclipse.xtext.builder.clustering.ClusteringBuilderState.doUpdate(ClusteringBuilderState.java:135)
	at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:116)
	at org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(XtextBuilder.java:287)
	at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:267)
	at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:161)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:142)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:232)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: org.xml.sax.SAXParseExceptionpublicId: platform:/resource/coffee_machine_pro/src/CoffeeMachine.sct; systemId: platform:/resource/coffee_machine_pro/src/CoffeeMachine.sct; lineNumber: 1; columnNumber: 2; Markup im Dokument vor dem Root-Element muss ordnungsgemäß formatiert sein.
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(Unknown Source)
	at org.eclipse.emf.ecore.xmi.impl.XMLLoadImpl.load(XMLLoadImpl.java:175)
	at org.eclipse.emf.ecore.xmi.impl.XMLResourceImpl.doLoad(XMLResourceImpl.java:261)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1518)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1297)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoad(ResourceSetImpl.java:259)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoadHelper(ResourceSetImpl.java:274)
	at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:265)
	at org.eclipse.xtext.resource.SynchronizedXtextResourceSet.getResource(SynchronizedXtextResourceSet.java:25)
	at org.eclipse.xtext.builder.resourceloader.SerialResourceLoader$1.next(SerialResourceLoader.java:47)
	... 19 more
`

### LINKED ISSUE #2087: Statechart diagrams cannot be saved when definition section is empty and pinned

### PR DESCRIPTION
**Pull request by robert-rudi**
Fixes #2044 
Fixes #2087

Introduced a new dialog when reloading a statechart model file which was valid and becomes invalid, i.e. by a vcs merge conflict.

### CODE REVIEW

File: plugins/org.yakindu.sct.domain/src/org/yakindu/sct/domain/extension/DomainRegistry.java [hunk 1]
```diff
@@ -117,7 +118,14 @@ public boolean apply(IDomain input) {
 
 	public static String determineDomainID(URI uri) {
 		if (URIConverter.INSTANCE.exists(uri, null)) {
-			return DomainIDParser.parse(uri);
+			try {
+				return DomainIDParser.parse(uri);
+			} catch (SAXException e) {
+				// the model is corrupt
+				return "";
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.domain/src/org/yakindu/sct/domain/extension/DomainRegistry.java]: should we return generic domain in such a case?

File: plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/resource/StextResource.java [hunk 2]
```diff
@@ -136,4 +141,22 @@ protected void parseTransition(Transition transition) {
 		}
 	}
 
+	@Override
+	public void doLoad(InputSource inputSource, Map<?, ?> options) throws IOException {
+		try {
+			super.doLoad(inputSource, options);
+		} catch (Exception e) {
+			// resource is corrupt
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/resource/StextResource.java]: quite catches in general are not a good idea, handle this properly

File: plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/resource/StextResource.java [hunk 3]
```diff
@@ -136,4 +141,22 @@ protected void parseTransition(Transition transition) {
...
+		}
+	}
+	
+	@Override
+	public void doLoad(InputStream inputStream, Map<?, ?> options) throws IOException {
+		try {
+			super.doLoad(inputStream, options);
+		} catch (Exception e) {
+			// resource is corrupt
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/resource/StextResource.java]: same here

**Comment by robert-rudi** [comment]
As mentioned in one of your comments, the DomainRegistry does now return the default domain, if the uri couldn't be parsed.
I removed the quite catches.

**Comment by andreasmuelder** [comment]
this will be separated into 2 different PRs

**Pull request closed without merging**